controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -17,48 +17,270 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/rancher-sandbox/cluster-api-provider-rke2/bootstrap/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// Rke2ControlPlaneSpec defines the desired state of Rke2ControlPlane
-type Rke2ControlPlaneSpec struct {
+// RKE2ControlPlaneSpec defines the desired state of RKE2ControlPlane
+type RKE2ControlPlaneSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Rke2ControlPlane. Edit rke2controlplane_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ServerConfig specifies configuration for the agent nodes.
+	//+optional
+	ServerConfig RKE2ServerConfig `json:"serverConfig,omitempty"`
+
+	// ManifestsConfigMapReference references a ConfigMap which contains Kubernetes manifests to be deployed automatically on the cluster
+	// Each data entry in the ConfigMap will be will be copied to a folder on the control plane nodes that RKE2 scans and uses to deploy manifests.
+	//+optional
+	ManifestsConfigMapReference corev1.ObjectReference `json:"manifestsConfigMapReference,omitempty"`
 }
 
-// Rke2ControlPlaneStatus defines the observed state of Rke2ControlPlane
-type Rke2ControlPlaneStatus struct {
+type RKE2ServerConfig struct {
+	// RKE2CommonNodeConfig is an inline struct that references common attribtes between agent and server nodes
+	v1alpha1.RKE2CommonNodeConfig `json:",inline"`
```

Comment:
If we use `bootstrapv1.RKE2ConfigSpec`  directly in RKE2ControlPlaneSpec then this could be removed from here (see previous comment first)

Reply:
Done!